tests/test_input_output.py

```diff
@@ -35,6 +38,59 @@ def test_vmec_input(tmpdir_factory):
         v.pop("output_path")
     assert all([equals(in1, in2) for in1, in2 in zip(vmec_inputs, desc_inputs)])
 
+    correct_file_path = ".//tests//inputs//input.DSHAPE_desc"
+    print(path)
+
+    # check DESC input file matches known correct one line-by-line
+    with open(correct_file_path) as f:
+        lines_correct = f.readlines()
+    with open(path) as f:
+        lines_direct = f.readlines()
+    with open(path_converted_file) as f:
+        lines_converted = f.readlines()
+    # skip first 3 lines as they have date and pwd info
+    for line1, line2 in zip(lines_correct[3:], lines_converted[3:]):
+        assert line1.strip() == line2.strip()
+    # skip first 4 here as the directly written file lacks a header
+    for line1, line2 in zip(lines_correct[4:], lines_direct):
+        assert line1.strip() == line2.strip()
+
+
+@pytest.mark.unit
+def test_write_desc_input_Nones(tmpdir_factory):
+    """Test converting writing DESC input file when an input tol is None."""
+    # tests how None is handled as a passed-in input to the input writer
+    input_path = "./tests/inputs/DSHAPE"
+    tmpdir = tmpdir_factory.mktemp("desc_inputs")
+    tmp_path = tmpdir.join("DSHAPE")
+    shutil.copyfile(input_path, tmp_path)
+    ir = InputReader(cl_args=[str(tmp_path)])
+
+    ftols_input_with_none = [1e-2, None, 1e-3]
```

Comment:
can you explain this expected behavior a bit? I thought if it's `None` it doesn't get written?

Reply:
True, but in this case the reader will assume that the 1e-3 is now used for the last 2 equilibria so I need to check that the 3 eq steps are 1e-2, 1e-3, 1e-3

in the string comparison part of the test I then check that the line itself only has 2 elements, 1e-2 and 1e-3

Reply:
I would prefer if `None` got replaced with the default tolerance, but since those are set by the Optimizer that isn't really possible.